### DISCUSSION TITLE
Netty TLS Symmetric Key

### DISCUSSION THREAD

**Discussion by Hcankaynak** [netty/netty#13441] [Q&A] [answered]
Hello everyone,

I have been using Netty for a while and I have become curious about something in Netty.

I have read that in TLS, after asymmetric encryption, the rest of the messages are encrypted with a symmetric key.

I have activated TLS in Netty and have handled the certificate and private key. Encryption works perfectly fine. However, I couldn't find information about what is used for symmetric key encryption. I didn't provide a symmetric key to the Netty SSL context, so I assume it is generated by default. Do you know the default settings, and can I insert a symmetric key into Netty?

Thank you for your time.

**Comment by chrisvest** [comment, marked as answer]
The symmetric cipher is negotiated during the TLS handshake, and the secret key is derived as well.
A cipher is picked such that it is supported by both the client and the server.

See for details: https://tls13.xargs.org/#client-hello/annotated
